apache2/re_operators.c

```diff
@@ -865,13 +865,8 @@ static int msre_op_validateHash_execute(modsec_rec *msr, msre_rule *rule, msre_v
         char *hmac = NULL, *valid = NULL;
         char *hash_link = NULL, *nlink = NULL;
 
-        if (strlen(pattern) > 252) {
-            *error_msg = apr_psprintf(msr->mp, "Request URI matched \"%.252s ...\" at %s.",
+        *error_msg = apr_psprintf(msr->mp, "Request URI matched \"%.252s ...\" at %.252s.",
```

Comment:
What will this print in cases where the pattern is less than 252? Blank string, 0s? %.252s

Reply:
Double this question. We need to make sure that if the input string is less than 252 characters long, it goes as is without extra padding or other unexpected effects.

If that does not work, perhaps something similar to https://stackoverflow.com/a/2641927 would help - we can take min(strlen(pattern), 252) and use it as a first argument.

Reply:
per my testing and the documentation, there is a way to provide min.max to string print, if min is not there, then it is not enforced